model/textparse/openmetricsparse.go

```diff
@@ -328,44 +327,54 @@ func (p *OpenMetricsParser) CreatedTimestamp() *int64 {
 		// All timestamps in OpenMetrics are Unix Epoch in seconds. Convert to milliseconds.
 		// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#timestamps
 		ct := int64(p.val * 1000.0)
-		p.setCTParseValues(ct, currFamilyLsetHash, currName, true, resetLexer)
+		p.setCTParseValues(ct, currHash, currName, true, resetLexer)
 		return &ct
 	}
 }
 
+var (
+	quantile = []byte{113, 117, 97, 110, 116, 105, 108, 101}
+	le       = []byte{108, 101}
+)
+
+// hashOffsets takes in the metric family name and hashes the offsets of the labels and values.
+func (p *OpenMetricsParser) hashOffsets(metricFamilyName []byte) uint64 {
```

Comment:
nice!

Reply:
It looks like from https://pprof.me/fc03179a8c2d4cb48bb6718f45c38d66/?profileType=profile%253Aalloc_objects%253Acount%253Aspace%253Abytes this function is still using ~30% of mem allocs of the CreatedTimestamp. Again, something we can iterate on, but I believe we can easily squeeze that by reusing `offsetsArr` (put that as argument or in parser and reset with `offsetsArr = offsetsArr[:0]` before use)

Reply:
pushing changes